ttnn/cpp/ttnn/operations/conv/conv_transpose2d/conv_transpose2d.cpp

```diff
@@ -0,0 +1,350 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "conv_transpose2d.hpp"
+#include <sys/types.h>
+#include <cstdint>
+#include "common/bfloat16.hpp"
+
+using namespace tt;
+namespace ttnn {
+namespace operations::conv {
+using sliding_window::SlidingWindowConfig;
+using sliding_window::ParallelConfig;
+
+namespace conv_transpose2d {
+
+template <typename T>
+Tensor _transform_weights_for_conv_transpose2d(
```

Comment:
oh host?

Reply:
Currently the weights preparation for Conv2d happens on host. It's a one-time operation, so made sense to implement on the host initially. But this is slow when loading large networks.  This must be ported to the device. 
Relevant issue : https://github.com/tenstorrent/tt-metal/issues/14080